src/simplify_reshapes.cpp

```diff
@@ -89,38 +89,23 @@ struct find_reshaper
 {
     auto matcher() const
     {
-        return match::name(reshaper_names())(
-            match::any_of[match::outputs()](match::name(reshaper_names())));
+        auto reshaper          = match::name(reshaper_names());
+        auto contiguous        = match::name("contiguous");
+        auto no_output_reshape = match::none_of[match::outputs()](reshaper);
+        auto input_reshape     = match::arg(0)(match::skip(contiguous)(reshaper));
+        auto input             = match::skip(reshaper, contiguous)(match::any().bind("x"));
+        return reshaper(no_output_reshape, input_reshape, input);
```

Comment:
This would at max find triply nested reshapes. We can probably do as earlier to walk back the graph inside the `apply()` to find the `start` of the nested reshapes and then do the replace.

Reply:
> This would at max find triply nested reshapes.

This will find an arbitrary number of nested reshapes, not just three. I updated the test to use 6 nested reshapes. 

> We can probably do as earlier to walk back the graph inside the apply() to find the start of the nested reshapes and then do the replace.

The `match::skip` already does this.